packages/template-retail-react-app/app/commerce-api/index.js

```diff
@@ -199,7 +199,7 @@ class CommerceAPI {
         // If the token is invalid (missing, past/nearing expiration), we issue
         //  a login call, which will attempt to refresh the token or get a new
         //  guest token. Once login is complete, we can proceed.
-        if (!isTokenValid(this.auth.authToken)) {
+        if (!this.auth.isTokenValid) {
             // NOTE: Login will update `this.auth.authToken` with a fresh token
```

Comment:
Just an observation and not something we need to change for this PR, but I'm wondering why we have this isTokenValid check both here and inside the Auth.login.

Reply:
@kevinxh can you comment about this ?

Reply:
the difference between adding here v.s. Auth.login is that, Auth.login is invoked on:
1. app start up
2. registered user login

Imagine a case where user lands on PWA and opens a second tab to navigate to SFRA, since the PWA page is not refreshed, it would still have the old values in memory, I think that's why Jainam added here to guard every single call, which makes sense.